modules/Dasboard/containers/teacher/DashboardContainer.tsx

```diff
@@ -1,13 +1,66 @@
-import React from "react";
+import { useEffect, useState } from "react";
 
+import { Container, LoadingOverlay } from "@mantine/core";
+import { notifications } from "@mantine/notifications";
+
+import { useGetClassroomsQuery } from "@/shared/redux/rtk-apis/classrooms/classrooms.api";
+
+import ClassroomList from "../../components/ClassroomCardList/ClassroomCardList";
 import CreateClassroomButton from "../../components/CreateClassroomButton/CreateClassroomButton";
 import Navbar from "../../components/Navbar/Navbar";
 
-const DashboardContainer = () => (
-  <div>
-    <Navbar />
-    <CreateClassroomButton />
-  </div>
-);
+const DashboardContainer: React.FC = () => {
+  const { data: classrooms, isLoading, error } = useGetClassroomsQuery();
+  const [hasAttemptedLoad, setHasAttemptedLoad] = useState(false);
+
+  useEffect(() => {
+    if (!isLoading) {
+      setHasAttemptedLoad(true);
+    }
+  }, [isLoading]);
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout | null = null;
+
+    if (error && hasAttemptedLoad) {
+      timer = setTimeout(() => {
+        if (!classrooms) {
+          notifications.show({
+            title: "Error",
+            message: "Failed to load classrooms. Please try again later.",
+            color: "red",
+            autoClose: 5000,
+          });
```

Comment:
You don't need to wrap all of this or the effect around this method. Just show this toast when error is true